File: fastapi_server/app/load_app_env.py
"""Load .env and vault file at startup."""

from dotenv import load_dotenv
from .vault_file import EnvStore, VaultFile
import os

# 0. Load .env file if exists
app_dir = os.path.dirname(os.path.abspath(__file__))
env_path = os.path.join(app_dir, ".env")
if os.path.exists(env_path):
    load_dotenv(env_path)
    print(f"  .env file: {env_path}")

# 1. Initialize Vault File with base64 file parsers
vault_file = os.getenv("VAULT_SECRET_FILE")
if vault_file:
    EnvStore.on_startup(
        vault_file,
        base64_file_parsers={
            'FILE_APP_ENV': lambda store: VaultFile.from_base64_auto(
                store.get('FILE_APP_ENV')
            )[0] if store.get('FILE_APP_ENV') else None  # [0] is parsed content, [1] is format
        }
    )
else:
    print("VAULT_SECRET_FILE env var not set, skipping EnvStore initialization")

File: fastify_server/src/load_app_env.mjs
/**
 * Load .env and vault file at startup.
 */

import { EnvStore, VaultFile } from "@internal/vault-file";
import dotenv from "dotenv";
import fs from "fs";
import path from "path";
import { fileURLToPath } from "url";

const __dirname = path.dirname(fileURLToPath(import.meta.url));
const projectRoot = path.resolve(__dirname, "..");

// 0. Load .env file if exists
const envPath = path.join(projectRoot, ".env");
if (fs.existsSync(envPath)) {
  dotenv.config({ path: envPath });
  console.log(`  .env file: ${envPath}`);
}

// 1. Initialize Vault File with base64 file parsers
const vaultFile = process.env.VAULT_SECRET_FILE;
if (vaultFile) {
  await EnvStore.onStartup(
    vaultFile,
    '.env*',
    false,
    {}, // computed definitions
    {
      FILE_APP_ENV: (store) => {
        const b64 = store.get('FILE_APP_ENV');
        return b64 ? VaultFile.fromBase64Auto(b64).content : null;  // .content is parsed data
      }
    }
  );
} else {
  console.log('VAULT_SECRET_FILE env var not set, skipping EnvStore initialization');
}

export { vaultFile };
